core/bootstrap/bootstrap.go

```diff
@@ -50,13 +50,31 @@ type BootstrapConfig struct {
 	// for the bootstrap process to use. This makes it possible for clients
 	// to control the peers the process uses at any moment.
 	BootstrapPeers func() []peer.AddrInfo
+
+	// SavePeersPeriod governs the periodic interval at which the node will
+	// attempt to save connected nodes to use as temporary bootstrap peers.
+	SavePeersPeriod time.Duration
+
+	// SaveConnectedPeersRatio controls the number peers we're saving compared
+	// to the target MinPeerThreshold. For example, if  MinPeerThreshold is 4,
+	// and we have a ratio of 5 we will save 20 connected peers.
+	//
+	// Note: one peer can have many addresses under its ID, so saving a peer
+	// might translate to more than one line in the config (following the above
+	// example that means TempBootstrapPeers may have more than 20 lines, but
+	// all those lines will be addresses of at most 20 peers).
+	SaveConnectedPeersRatio   int
+	SaveTempPeersForBootstrap func(context.Context, []peer.AddrInfo)
+	LoadTempPeersForBootstrap func(context.Context) []peer.AddrInfo
 }
 
 // DefaultBootstrapConfig specifies default sane parameters for bootstrapping.
 var DefaultBootstrapConfig = BootstrapConfig{
-	MinPeerThreshold:  4,
-	Period:            30 * time.Second,
-	ConnectionTimeout: (30 * time.Second) / 3, // Perod / 3
+	MinPeerThreshold:        4,
+	Period:                  30 * time.Second,
+	ConnectionTimeout:       (30 * time.Second) / 3, // Perod / 3
+	SavePeersPeriod:         1 * time.Hour,
```

Comment:
@lidel the way it is currently implemented, it will save peers right after the first bootstrap. Then, it runs every interval. So we don't need to wait 1 hour. I will check.